docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -832,6 +832,23 @@ Cython code.  Here is the list of currently supported directives:
     set to ``None``. Otherwise a check is inserted and the
     appropriate exception is raised. This is off by default for
     performance reasons.  Default is False.
+    
+``threadsafe_variable_access`` (off / refcounted, full)
+    Defines how Cython will try to ensure thread-safety in freethreading
+    Python builds.  If set to ``off`` (the fastest) then Cython will not
+    do anything to ensure thread-safety.  If set to ``refcounted`` then
+    Cython will ensure the internal consistency of reference counting
+    for Python objects and typed memoryviews.  If set to ``full`` then
+    Cython will also guard access to C types (int, float, etc) so they
+    should be read and written atomically.
```

Comment:
This sentence reads more costly than it probably is, right? You're not specially guarding all C operations on simple numeric C types, are you?

Reply:
> This sentence reads more costly than it probably is, right? You're not specially guarding all C operations on simple numeric C types, are you?

Yes - that's largely it.  Although most function-local variables are excluded because there's no way that multiple threads have simultaneous access.

I'm assuming it will be fairly expensive and isn't really helpful in keeping the interpreter state consistent, so it wasn't on by default.

I could definitely be persuaded that it isn't a useful level though.